unit_tests/test_config.py

```diff
@@ -674,6 +674,32 @@ def test_22_verify_same_stress_cmds_for_all_tenants(self):
         self.assertEqual(tenant_stress_cmds_0['stress_read_cmd'], tenant_stress_cmds_1['stress_read_cmd'])
         self.assertEqual(tenant_stress_cmds_0['prepare_write_cmd'], tenant_stress_cmds_1['prepare_write_cmd'])
 
+    def test_23_nested_values(self):
```

Comment:
Anything in particular you see missing ? otherwise whom would come with a more complex usage to it, would need to add tests for it.

Reply:
Attempt to set the same key twice. There is no possible case when we want to redefine the line in the same configuration file. So, we should raise error in such a case, because most likely it is typo that will lead to the unexpected results.
And now, as I understand, we will silently redefine values and apply only latest one.

Reply:
set the same key twice where ? 
- in a yaml ? that's a broken yaml and it can't be read.
- in multiple yamls, be design we overwrite keys (and it's not related to this change)
- in environment variable ?

you'll have to be a bit more concrete about what is the concern

Reply:
I was referring to the first variant - definition of the same key twice as part of the same yaml.
And such case is not covered in unit tests. So, may think that it cannot be read but actually it is not tested.

Reply:
So you say we don't cover broken yaml in our tests, that's might be correct. but why it's relate to this change specifically ? i.r. it's not the first time we introduce reading yaml for configuration.

Reply:
This a known issue with pyyaml:
https://github.com/yaml/pyyaml/issues/41

Suggestions are to switch to something else. but again, not related to changes here.